src/analog/m2kanalogin_impl.cpp

```diff
@@ -510,82 +500,69 @@ std::vector<std::pair<std::string, std::pair<double, double>>> M2kAnalogInImpl::
+	m_adc_hw_offset_raw.at(channel) = m_ad5625_dev->setLongValue(channel + 2, hw_offset_raw, "raw", true);
 	m_trigger->setCalibParameters(channel, getScalingFactor(channel), m_adc_hw_vert_offset.at(channel));
 }
 
 double M2kAnalogInImpl::getVerticalOffset(ANALOG_IN_CHANNEL channel)
 {
-	double gain = 1.3;
-	double vref = 1.2;
-	int channel_number = 2 + channel;
-	double hw_range_gain = getValueForRange(m_input_range.at(channel));
-	m_adc_hw_vert_offset_raw.at(channel) = m_ad5625_dev->getLongValue(channel_number, "raw", true);
-
-	double vert_offset = (m_adc_hw_vert_offset_raw.at(channel) - m_adc_calib_offset.at(channel)) *
-			2.693 * vref / ((1 << 12) * hw_range_gain * gain);
-	m_adc_hw_vert_offset.at(channel) = vert_offset;
-	return vert_offset;
+	return m_adc_hw_vert_offset.at(channel);
 }
 
 int M2kAnalogInImpl::getOversamplingRatio()
 {
-	return getLongValue("oversampling_ratio");
+	return m_oversampling_ratio;
 }
 
 int M2kAnalogInImpl::getOversamplingRatio(unsigned int chn_idx)
 {
-	return getLongValue(chn_idx, "oversampling_ratio");
+	return m_oversampling_ratio;
 }
 
 int M2kAnalogInImpl::setOversamplingRatio(int oversampling_ratio)
 {
-	return setLongValue(oversampling_ratio, "oversampling_ratio");
+	m_oversampling_ratio = m_m2k_adc->setLongValue(oversampling_ratio, "oversampling_ratio");
+	return m_oversampling_ratio;
 }
 
 int M2kAnalogInImpl::setOversamplingRatio(unsigned int chn_idx, int oversampling_ratio)
 {
-	return setLongValue(chn_idx, oversampling_ratio, "oversampling_ratio");
+	m_oversampling_ratio = m_m2k_adc->setLongValue(oversampling_ratio, "oversampling_ratio");
+	return m_oversampling_ratio;
 }
 
 
 double M2kAnalogInImpl::getSampleRate()
 {
-	return getDoubleValue("sampling_frequency");
+	return m_samplerate;
 }
 
 double M2kAnalogInImpl::getSampleRate(unsigned int chn_idx)
 {
-	return getDoubleValue(chn_idx, "sampling_frequency");
+	return m_samplerate;
 }
 
 double M2kAnalogInImpl::setSampleRate(double samplerate)
 {
-	return setDoubleValue(samplerate, "sampling_frequency");
+	m_samplerate = m_m2k_adc->setLongValue((int)samplerate, "sampling_frequency");
 	m_trigger->setCalibParameters(0, getScalingFactor(0), m_adc_hw_vert_offset.at(0));
 	m_trigger->setCalibParameters(1, getScalingFactor(1), m_adc_hw_vert_offset.at(1));
+	return m_samplerate;
 }
 
 double M2kAnalogInImpl::setSampleRate(unsigned int chn_idx, double samplerate)
 {
-	return setDoubleValue(chn_idx, samplerate, "sampling_frequency");
-	m_trigger->setCalibParameters(0, getScalingFactor(0), m_adc_hw_vert_offset.at(0));
-	m_trigger->setCalibParameters(1, getScalingFactor(1), m_adc_hw_vert_offset.at(1));
+	m_samplerate = m_m2k_adc->setLongValue((int)samplerate, "sampling_frequency");
+	m_trigger->setCalibParameters(chn_idx, getScalingFactor(chn_idx), m_adc_hw_vert_offset.at(chn_idx));return m_samplerate;
 }
 
 double M2kAnalogInImpl::getFilterCompensation(double samplerate)
 {
-	double compensation = 0.0;
+	double compensation;
```

Comment:
Is there any reason to not have this variable initialized here? http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-always